tools/test.py

```diff
@@ -1505,7 +1526,11 @@ def Main():
 
   result = None
   def DoSkip(case):
-    return SKIP in case.outcomes or SLOW in case.outcomes
+    if SKIP in case.outcomes or SLOW in case.outcomes:
+      return True
+    if FLAKY in case.outcomes and options.flaky_tests == SKIP:
```

Comment:
Minor nit. We can simply return the result of this expression `return FLAKY in case.outcomes and options.flaky_tests == SKIP`